syntax = "proto3";

package myChatMessage; 

enum ChatMessageType
{
    SERVER_PING = 0;
    SERVER_MESSAGE = 1;

    ALL_MESSAGE = 2;
    
    WHISPER_MESSAGE = 3;
    
    PARTY_MESSAGE = 4;
    PARTY_CREATE = 41;
    PARTY_DELETE = 42;
    PARTY_JOIN = 43; 
    PARTY_LEAVE = 44;

    FRIEND_REQUEST = 50;
    FRIEND_ACCEPT = 51;
    FRIEND_REJECT = 52;

    LOGIN_MESSAGE = 70;

    ERROR_MESSAGE = 99;
}

message ChatMessage {
    ChatMessageType messageType = 1;
    string sender = 2;
    string receiver = 3;
    string content = 4;    
}